runtime/near-vm-runner/src/imports.rs

```diff
@@ -429,61 +429,37 @@ pub(crate) mod wasmer2 {
 #[cfg(feature = "wasmtime_vm")]
 pub(crate) mod wasmtime {
     use super::str_eq;
-    use near_vm_logic::{ProtocolVersion, VMLogic, VMLogicError};
-    use std::cell::{RefCell, UnsafeCell};
-    use std::ffi::c_void;
+    use crate::wasmtime_runner::VMCtx;
+    use near_vm_logic::{ProtocolVersion, VMLogicError};
+    use std::cell::RefCell;
 
     thread_local! {
-        static CALLER_CONTEXT: UnsafeCell<*mut c_void> = UnsafeCell::new(0 as *mut c_void);
         static EMBEDDER_ERROR: RefCell<Option<VMLogicError>> = RefCell::new(None);
```

Comment:
Unrelated, but we can get rid of this because `impl From<Box<dyn std::error::Error + Send + Sync>> for wasmtime::Trap` and `impl std::error::Error for Trap`, with `Error::source` I think exposing the underlying error.

Reply:
I belive we need to send the following patch to wasmtime for that to work: 

```diff
diff --git a/crates/wasmtime/src/trap.rs b/crates/wasmtime/src/trap.rs
index 87d183200..7e9e311bb 100644
--- a/crates/wasmtime/src/trap.rs
+++ b/crates/wasmtime/src/trap.rs
@@ -379,7 +379,7 @@ impl fmt::Display for Trap {
 impl std::error::Error for Trap {
     fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
         match &self.inner.reason {
-            TrapReason::Error(e) => e.source(),
+            TrapReason::Error(e) => Some(&**e),
             TrapReason::I32Exit(_) | TrapReason::Message(_) | TrapReason::InstructionTrap(_) => {
                 None
             }
```

probably won't get into looking into that further.